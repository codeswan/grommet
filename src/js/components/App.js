// (C) Copyright 2014-2016 Hewlett Packard Enterprise Development LP

import React, { Component, PropTypes } from 'react';
import classnames from 'classnames';
import { getCurrentLocale } from '../utils/Locale';
import SkipLinks from './SkipLinks';

import CSSClassnames from '../utils/CSSClassnames';

const CLASS_ROOT = CSSClassnames.APP;

let supportedLocales = ['en-US', 'pt-BR'];

function localesSupported() {
  return global.Intl && supportedLocales.every(function (locale) {
    return Intl.NumberFormat.supportedLocalesOf(locale)[0] === locale &&
            Intl.DateTimeFormat.supportedLocalesOf(locale)[0] === locale;
  });
}

if (! localesSupported()) {
  require('intl');
  require('intl/locale-data/jsonp/en-US.js');
  require('intl/locale-data/jsonp/pt-BR.js');
  Intl.NumberFormat = IntlPolyfill.NumberFormat;
  Intl.DateTimeFormat = IntlPolyfill.DateTimeFormat;
}

export default class App extends Component {
  constructor(props, context) {
    super(props, context);

    this.state = {
      lang: 'en-US'
    };
  }

  componentDidMount() {
    var lang = getCurrentLocale();
    if (this.props.lang) {
      lang = this.props.lang;
    }
    if (!document.documentElement.getAttribute('lang')) {
      document.documentElement.setAttribute('lang', lang);
    }
    this.setState({lang: lang});
  }

  render() {
    const { centered, children, className, inline } = this.props;
    const { lang } = this.state;

    const classes = classnames(
      'grommet',
      className,
      CLASS_ROOT, {
        [`${CLASS_ROOT}--centered`]: centered,
        [`${CLASS_ROOT}--inline`]: inline
      }
    );

    return (
      <div lang={lang} className={classes}>
        {children}
        <SkipLinks />
        <div className={`${CLASS_ROOT}__announcer`} aria-live='polite' />
      </div>
    );
  }
}

App.defaultProps = {
  centered: true
};

App.propTypes = {
  centered: PropTypes.bool
};
