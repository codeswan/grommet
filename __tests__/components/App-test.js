// (C) Copyright 2014-2016 Hewlett Packard Enterprise Development LP

import React from 'react';
import renderer from 'react/lib/ReactTestRenderer';

import App from '../../src/js/components/App';

// needed because this:
// https://github.com/facebook/jest/issues/1353
jest.mock('react-dom');

describe('App', () => {
  it('has correct default options', () => {
    const component = renderer.create(
      <App><h2>App Body</h2></App>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('has correct inline=true rendering', () => {
    const component = renderer.create(
      <App inline={true} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('has correct className rendering', () => {
    const component = renderer.create(
      <App className='testing' />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
