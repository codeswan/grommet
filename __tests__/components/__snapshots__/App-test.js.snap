exports[`App has correct className rendering 1`] = `
<div
  className="grommet testing grommetux-app grommetux-app--centered"
  lang="en-US">
  <span
    style={
      Object {
        "display": "none"
      }
    } />
  <div
    aria-live="polite"
    className="grommetux-app__announcer" />
</div>
`;

exports[`App has correct default options 1`] = `
<div
  className="grommet grommetux-app grommetux-app--centered"
  lang="en-US">
  <h2>
    App Body
  </h2>
  <span
    style={
      Object {
        "display": "none"
      }
    } />
  <div
    aria-live="polite"
    className="grommetux-app__announcer" />
</div>
`;

exports[`App has correct inline=true rendering 1`] = `
<div
  className="grommet grommetux-app grommetux-app--centered grommetux-app--inline"
  lang="en-US">
  <span
    style={
      Object {
        "display": "none"
      }
    } />
  <div
    aria-live="polite"
    className="grommetux-app__announcer" />
</div>
`;
